packages/playground/src/components/withdraw_dialog.vue

```diff
@@ -13,33 +13,49 @@
         </v-card-text>
         <v-card-text>
-          <v-form v-model="isValidSwap">
-            <v-text-field
-              v-model="target"
-              :label="selectedName?.charAt(0).toUpperCase() + selectedName!.slice(1) + ' Target Wallet Address'"
-              :error-messages="targetError"
-              :disabled="validatingAddress"
-              :loading="validatingAddress"
-              :rules="[() => !!target || 'This field is required', swapAddressCheck]"
+          <FormValidator v-model="valid">
+            <InputValidator
+              :value="target"
+              :rules="[validators.required('This field is required'), () => swapAddressCheck()]"
+              :async-rules="[validateAddress]"
+              #="{ props: validationProps }"
             >
-            </v-text-field>
-            <v-text-field
-              @paste.prevent
-              label="Amount (TFT)"
-              v-model="amount"
-              type="number"
-              onkeydown="javascript: return event.keyCode == 69 || /^\+$/.test(event.key) ? false : true"
+              <v-text-field
+                v-bind="{ ...validationProps }"
+                v-model="target"
+                :label="selectedName?.charAt(0).toUpperCase() + selectedName!.slice(1) + ' Target Wallet Address'"
+                :disabled="validatingAddress"
+                :loading="validationProps.loading"
+              >
+              </v-text-field>
+            </InputValidator>
+            <InputValidator
+              :value="amount"
+              #="{ props: validationProps }"
               :rules="[
-                () => !!amount || 'This field is required',
-                () =>
-                  (amount.toString().split('.').length > 1 ? amount.toString().split('.')[1].length <= 3 : true) ||
-                  'Amount must have 3 decimals only',
-                () => amount >= 2 || 'Amount should be at least 2 TFT',
-                () => amount < freeBalance! || 'Amount cannot exceed balance',
-              ]"
+              validators.required('This field is required'),
+              validators.min('Amount should be at least 2 TFT', 2),
+              validators.max( 'Amount cannot exceed balance',freeBalance!),
+              () => {
+                if(!(amount.toString().split('.').length > 1 ? amount.toString().split('.')[1].length <= 3 : true)){
+                  return {
+                    message:  'Amount must have 3 decimals only'
+                  }
+                }
+              }
+            ]"
             >
-            </v-text-field>
-          </v-form>
+              <v-text-field
+                v-bind="{ ...validationProps }"
+                @paste.prevent
+                label="Amount (TFT)"
+                v-model="amount"
+                type="number"
+                onkeydown="javascript: return event.keyCode == 69 || /^\+$/.test(event.key) ? false : true"
```

Comment:
what is this code used for .... I'm pretty sure we can remove it and code should last working as before

Reply:
after discussion with @MohamedElmdary we agreed that this line is functional and it prevents user from typing `e` and +